api/src/test/java/org/openmrs/api/impl/DiagnosisServiceImplTest.java

```diff
@@ -0,0 +1,209 @@
+		Assert.assertEquals(3, diagnoses.size());
+		Assert.assertEquals("68802cce-6880-17e4-6880-a68804d22fb7", diagnoses.get(0).getUuid());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, diagnoses.get(0).getCertainty());
+		Assert.assertEquals(new Integer(1), diagnoses.get(0).getDiagnosisId());
+		Assert.assertEquals(new Integer(2), diagnoses.get(0).getPatient().getPatientId());
+		Assert.assertEquals("688804ce-6880-8804-6880-a68804d88047", diagnoses.get(2).getUuid());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, diagnoses.get(2).getCertainty());
+		Assert.assertEquals(new Integer(3), diagnoses.get(2).getDiagnosisId());
+		Assert.assertEquals(new Integer(2), diagnoses.get(2).getPatient().getPatientId());
+	}
+
+	/**
+	 * @see DiagnosisService#getUniqueDiagnoses(Patient, Date) 
+	 */
+	@Test
+	public void getUniqueDiagnoses_shouldGetUniqueDiagnosesOfPatient(){
+		Patient patient = patientService.getPatient(2);
+		List<Diagnosis> diagnoses = diagnosisService.getUniqueDiagnoses(patient, new Date(0));
+		
+		Assert.assertEquals("68802cce-6880-17e4-6880-a68804d22fb7", diagnoses.get(0).getUuid());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, diagnoses.get(0).getCertainty());
+		Assert.assertEquals(new Integer(1), diagnoses.get(0).getDiagnosisId());
+		Assert.assertEquals(new Integer(2), diagnoses.get(0).getPatient().getPatientId());
+		Assert.assertEquals(1, diagnoses.size());
+	}
+
+	/**
+	 * @see DiagnosisService#voidDiagnosis(Diagnosis, String)
+	 */
+	@Test
+	public void voidDiagnosis_shouldVoidDiagnosisSuccessfully(){
+		String voidReason = "void reason";
+		String uuid = "688804ce-6880-8804-6880-a68804d88047";
+		Diagnosis nonVoidedDiagnosis = diagnosisService.getDiagnosisByUuid(uuid);
+		Assert.assertFalse(nonVoidedDiagnosis.getVoided());
+		Assert.assertNull(nonVoidedDiagnosis.getVoidedBy());
+		Assert.assertNull(nonVoidedDiagnosis.getVoidReason());
+		diagnosisService.voidDiagnosis(nonVoidedDiagnosis, voidReason);
+		Diagnosis voidedDiagnosis = diagnosisService.getDiagnosisByUuid(uuid);
+		Assert.assertTrue(voidedDiagnosis.getVoided());
+		Assert.assertNotNull(voidedDiagnosis.getVoidedBy());
+		Assert.assertNotNull(voidedDiagnosis.getDateVoided());
+		Assert.assertEquals(voidReason, voidedDiagnosis.getVoidReason());
+		Assert.assertEquals(Context.getAuthenticatedUser(), voidedDiagnosis.getVoidedBy());
+	}
+
+	/**
+	 * @see DiagnosisService#unvoidDiagnosis(Diagnosis)
+	 */
+	@Test
+	public void unvoidDiagnosis_shouldUnvoidDiagnosisSuccessfully(){
+		String uuid = "77009cce-8804-17e4-8804-a68804d22fb7";
+		Diagnosis voidedDiagnosis = diagnosisService.getDiagnosisByUuid(uuid);
+		Assert.assertTrue(voidedDiagnosis.getVoided());
+		Assert.assertNotNull(voidedDiagnosis.getVoidReason());
+		Assert.assertNotNull(voidedDiagnosis.getDateVoided());
+		Assert.assertNotNull(voidedDiagnosis.getVoidedBy());
+		
+		Diagnosis unVoidedDiagnosis= diagnosisService.unvoidDiagnosis(voidedDiagnosis);
```

Comment:
Can you instead fetch this again with diagnosisService.getDiagnosisByUuid()

Reply:
Okay. Done